packages/google_adsense/example/integration_test/experimental_ad_unit_widget_test.dart

```diff
@@ -19,6 +19,17 @@ import 'js_interop_mocks/adsense_test_js_interop.dart';
 const String testClient = 'test_client';
 const String testSlot = 'test_slot';
 
+class CallbackTracker {
+  void Function() createCallback() {
+    final int callbackIndex = _callbackStates.length;
+    _callbackStates.add(false);
+    return () => _callbackStates[callbackIndex] = true;
+  }
+
+  bool get allCalled => _callbackStates.every((bool state) => state);
```

Comment:
What about something like:

`Future<bool> allCalled => Future.wait(_callbackFutures).timeout(const Duration(seconds: 10));`

The implementation of createCallback would insert a `Future<bool>` (or `Future<void>`, probably doesn't matter) into the `_callbackStates`, and the function would just `.complete` the completer for that future.

With the timeout, we can ensure the "busy loop" pumping frames doesn't keep pumping frames until the test times out (which can take several minutes to do so), and tests should fail kinda fast. (I think that if left unhandled, the `timeout` just throws a TimeoutException.)